pkg/xds/factory.go

```diff
@@ -0,0 +1,60 @@
+package xds
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/golang/protobuf/ptypes/duration"
+	jsoniter "github.com/json-iterator/go"
+	mv2 "mosn.io/mosn/pkg/config/v2"
+	"mosn.io/mosn/pkg/types"
+	v2conv "mosn.io/mosn/pkg/xds/v2/conv"
+	v3conv "mosn.io/mosn/pkg/xds/v3/conv"
+)
+
+var json = jsoniter.ConfigCompatibleWithStandardLibrary
+
+func duration2String(duration *duration.Duration) string {
+	d := time.Duration(duration.Seconds)*time.Second + time.Duration(duration.Nanos)*time.Nanosecond
+	x := fmt.Sprintf("%.9f", d.Seconds())
+	x = strings.TrimSuffix(x, "000")
+	x = strings.TrimSuffix(x, "000")
+	return x + "s"
+}
+
+// Client XdsClient
+type Client interface {
+	Start(config *mv2.MOSNConfig) error
+	Stop()
+}
+
+// NewClient build xds Client
+func NewClient() Client {
+	switch types.XdsVersion {
```

Comment:
wouldn't it (`XdsVersion`) be better for this configuration to support configurability during the v2 and v3 transition?